src/core/backbone_router/multicast_listeners_table.hpp

```diff
@@ -0,0 +1,226 @@
+         */
+        const TimeMilli GetExpireTime(void) const { return mExpireTime; }
+
+    private:
+        void SetAddress(const Ip6::Address &aAddress) { mAddress = aAddress; }
+        void SetExpireTime(TimeMilli aExpireTime) { mExpireTime = aExpireTime; }
+
+        static bool IsAfter(const Listener &aOne, const Listener &aOther)
+        {
+            return aOne.GetExpireTime() > aOther.GetExpireTime();
+        }
+
+        Ip6::Address mAddress;
+        TimeMilli    mExpireTime;
+    };
+
+    /**
+     * This constructor initializes the Multicast Listeners Table.
+     *
+     * @param[in] aInstance  A reference to the OpenThread instance.
+     *
+     */
+    explicit MulticastListenersTable(Instance &aInstance)
+        : InstanceLocator(aInstance)
+        , mNumValidListeners(0)
+    {
+    }
+
+    /**
+     * This method adds an Ip6 multicast address with a given expire time.
+     *
+     * @param[in] aAddress     The Ip6 multicast address.
+     * @param[in] aExpireTime  The expire time.
+     *
+     * @retval OT_ERROR_NONE          If the Ip6 multicast address was successfully added.
+     * @retval OT_ERROR_INVALID_ARGS  If the Ip6 multicast address was invalid.
+     * @retval OT_ERROR_NO_BUFS       No space available to save the Ip6 multicast address.
+     *
+     */
+    otError Add(const Ip6::Address &aAddress, TimeMilli aExpireTime);
+
+    /**
+     * This method removes a given Ip6 multicast address.
+     *
+     * @param[in] aAddress  The Ip6 multicast address to remove.
+     *
+     */
+    void Remove(const Ip6::Address &aAddress);
+
+    /**
+     * This method removes expired Ip6 multicast addresses.
+     *
+     */
+    void Expire(void);
+
+    /**
+     * This method counts the number of valid Ip6 multicast addresses.
+     * @return
+     */
+    uint16_t Count(void) const { return mNumValidListeners; }
+
+    /**
+     * This method enables range-based `for` loop iteration over all Listener entries.
+     *
+     * This method should be used as follows:
+     *
+     *     for (Child &child : aChildTable.Iterate(aFilter)) { ... }
+     *
+     * @param[in] aFilter  A child state filter.
+     *
+     * @returns An IteratorBuilder instance.
+     *
+     */
+    IteratorBuilder Iterate(void) { return IteratorBuilder(GetInstance()); }
+
+    /**
+     * This method clears the Multicast Listeners Table.
+     *
+     */
+    void Clear(void);
+
+    /**
+     * This method sets the Multicast Listener callback.
+     *
+     * @param aCallback  The callback function.
+     */
+    void SetCallback(otBackboneRouterMulticastListenerCallback aCallback, void *aContext);
+
+#if OPENTHREAD_CONFIG_REFERENCE_DEVICE_ENABLE
+    otError GetNext(otBackboneRouterMulticastListenerIterator &aIterator,
+                    otBackboneRouterMulticastListenerInfo &    aListenerInfo);
+#endif
+
+private:
+    enum
+    {
+        kMulticastListenersTableSize = OPENTHREAD_CONFIG_MAX_MULTICAST_LISTENERS
+    };
```

Comment:
assert that `kMulticastListenersTableSize` must be no less than 75 which is the minimal requirement defined in Conformance Specification?